include/maidsafe/common/rsa.h

```diff
@@ -85,10 +86,15 @@ PlainText Decrypt(const CipherText& data, const PrivateKey& private_key);
 
 Signature Sign(const PlainText& data, const PrivateKey& private_key);
 
+std::vector<byte> Sign(const std::vector<byte>& data, const PrivateKey& private_key);
```

Comment:
I guess I don't feel very strongly, but I think we lose a bit of type-safety here, and I'd choose type-safety over efficiency until we're at the profiling stage.

Also, the `Signature` type doesn't just give us type-safety; it's checking the bounds too.  We could achieve something similar here by maybe using a `std::array<byte, Keys::kSignatureByteSize>` rather than a `std::vector<byte>`, but it's not as robust in terms of checking the bounds.